flink-filesystems/flink-azure-fs-hadoop/src/main/java/org/apache/flink/fs/azurefs/AbstractAzureFSFactory.java

```diff
@@ -75,18 +76,27 @@ public void configure(Configuration config) {
     @Override
     public FileSystem create(URI fsUri) throws IOException {
         checkNotNull(fsUri, "passed file system URI object should not be null");
-        LOG.info("Trying to load and instantiate Azure File System");
+        LOG.info("Trying to load and instantiate Azure File System for {}", fsUri);
         return new HadoopFileSystem(createInitializedAzureFS(fsUri, flinkConfig));
     }
 
-    // uri is of the form: wasb(s)://yourcontainer@youraccount.blob.core.windows.net/testDir
+    // uri is of the form: wasb(s)://yourcontainer@youraccount.blob.core.windows.net/testDir (or)
+    // abfs(s):////yourcontainer@youraccount.dfs.core.windows.net/testDir
     private org.apache.hadoop.fs.FileSystem createInitializedAzureFS(
             URI fsUri, Configuration flinkConfig) throws IOException {
         org.apache.hadoop.conf.Configuration hadoopConfig = configLoader.getOrLoadHadoopConfig();
-
-        org.apache.hadoop.fs.FileSystem azureFS = new NativeAzureFileSystem();
-        azureFS.initialize(fsUri, hadoopConfig);
-
-        return azureFS;
+        String scheme = fsUri.getScheme();
+
+        if (scheme.startsWith("wasb")) {
```

Comment:
How about adding a 
```abstract org.apache.hadoop.fs.FileSystem createAzureFS();```
method to this factory and avoid this if-else?
Then the abstract factory doesn't need to know the scheme of the implementations.

Reply:
I have updated the PR, Please let me know if this looks OK. Thank you :)

Reply:
From code structure, it now looks much cleaner. We could go one step further and pull the `initialize` back into this method. Then all `createAzureFS` are really just one liners.

Reply:
Great idea, That makes a lot of sense :).